File: Leetcode/169_Majority_Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int, int> m;
        int n = nums.size();
        for (auto num: nums) {
            if (++m[num] > n / 2) return num;
        }
        return nums[0];
    }
};

File: Leetcode/169_Majority_Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        m = dict()
        c = len(nums) / 2
        for n in nums:
            if n not in m:
                m[n] = 1
            else:
                m[n] += 1
            if m[n] > c:
                return n
        return nums[0]